icu4c/source/common/loclikely.cpp

```diff
@@ -1408,7 +1408,9 @@ ulocimp_getRegionForSupplementalData(const char *localeID, UBool inferRegion,
         }
     }
 
-    rgBuf[rgLen] = 0;
+    if (rgLen >= 0) {
+        rgBuf[rgLen] = 0;
+    }
     uprv_strncpy(region, rgBuf, regionCapacity);
```

Comment:
There is no way that rgLen could be negative here.
However, I believe that there is a chance that rgLen == ULOC_RG_BUFLEN (from a bogus locale ID via uloc_getCountry()) in which case we must not write the terminating NUL here.
Also, we don't need the NUL if we use memcpy(), and u_terminateChars() will, well, terminate the region buffer if there is room.

```suggestion
    if (U_SUCCESS(*status) && rgLen <= regionCapacity) {
        uprv_memcpy(region, rgBuf, rgLen);
    }
```

Reply:
This part of the code does not seem to exist in current `main` anymore, so this can be omitted.